assets/js/sensei-modal.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * @modulw sensei-modal
+ * @description Adds a basic suport for modals via "data-sensei-modal-*" attribures on HTML elements.
+ *
+ * @usage
+ * The Sensei Modal consists of four basic elements: open, close, content and overlay.
+ * Each of those elements should be denoted with HTML attributes:
+ * - data-sensei-modal-open
+ * - data-sensei-modal-close
+ * - data-sensei-modal-content
+ * - data-sensei-modal-overlay
+ * The attribute values should be a unique id for each modal. Should look like this:
+ * ```html
+ * <button data-sensei-modal-open="123">Open Modal</button>
+ * <div data-sensei-modal-overlay="123"></div>
+ * <div data-sensei-modal-content="123">
+ *   <h1>Hello Modal!</h1>
+ *   <button data-sensei-modal-close="123">Close Modal</button>
+ * </div>
+ * ```
+ */
+
+/**
+ * Internal dependencies
+ */
+import { onWindowLoad } from '../shared/helpers/DOM';
+
+/**
+ * Opens the modal.
+ * @param {string} modalId The id of the modal.
+ */
+const openModal = ( modalId ) => ( ev ) => {
+	ev?.preventDefault();
+
+	document
+		.querySelector( `[data-sensei-modal-content="${ modalId }"]` )
+		?.setAttribute( 'data-sensei-modal-content-is-open', '' );
+	document
+		.querySelector( `[data-sensei-modal-overlay="${ modalId }"]` )
+		?.setAttribute( 'data-sensei-modal-overlay-is-open', '' );
+};
+
+/**
+ * Closes the modal.
+ * @param {string} modalId The id of theh modal.
+ */
+const closeModal = ( modalId ) => ( ev ) => {
+	ev?.preventDefault();
+
+	document
+		.querySelector( `[data-sensei-modal-overlay="${ modalId }"]` )
+		?.removeAttribute( 'data-sensei-modal-overlay-is-open' );
+	document
+		.querySelector( `[data-sensei-modal-content="${ modalId }"]` )
+		?.removeAttribute( 'data-sensei-modal-content-is-open' );
+};
+
+/**
+ * Attach modal events.
+ */
+function attachModalEvents() {
+	// Attach open events.
+	document
+		.querySelectorAll( '[data-sensei-modal-open]' )
```

Comment:
I think it's better to simplify the modal "api" instead of simplifying the internals of the modal "library". If we require to put all modal parts inside one element, then we impose extra limitations.

For instance with current approach you can use modals without overlay and close elements and it still works as expected. You can still open the modal and close the modal by pressing escape key. I think it's more versatile this way.

At least that was my thinking initially. WDYT?

Reply:
I'd argue it makes the "api" simpler, since you don't have to attach an ID to all the elements inside to make it work. It'd still be fine to leave out parts, the mounting function doesn't have to stop if it can't find a close button or something.

Reply:
I think you're right. I simplified the implementation and now it works the way you suggested. Let me know what you think. 6dcc3b9cb7c0fb02128b976549c1cc1a61e6b379